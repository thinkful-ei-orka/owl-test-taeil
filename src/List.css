
.participant {
  display: flex;
  margin: 10px 0;
}
.participant img {
  width: 40px;
  height: 40px;
  background: #ddd;
  border-radius: 50%;
  margin-right: 10px;
}
.participant div {

}
.participant h6 {
  margin: 0;
  font-size: 16px;
  margin-top: 3px;
}


.status {
  display: inline-block;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin-right: 5px;
}
.participant.session-true .status {
  background: #65D482;
}
.participant.session-true .off-stage {
  display: none;
}
.participant.session-false .status {
  background: rgb(218, 120, 120);
}
.participant.session-false .on-stage {
  display: none;
}
.participant.session-false .off-stage {
  display: inline;
}
.participant.stage-true {

}
.participant.stage-off {

}

.on-stage, .off-stage, .boot {
  color: #adadad;
  font-size: 14px;
}
