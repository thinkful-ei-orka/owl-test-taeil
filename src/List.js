import React from 'react';

import './List.css';
import data from './data';

function List() {
  let sessionParticipants = [];
  let leftParcipants = [];

  data.participants.forEach((participant) => {
    let html =
      <div className={'participant session-' + participant.inSession + ' stage-' + participant.onStage} key={participant.id}>
        <img src={participant.avatar} alt={participant.name + ' avatar'} />
        <div>
          <h6>{participant.name}</h6>
          <div>
            <span className="status"></span>
            <span className="on-stage">on stage</span>
            <span className="off-stage">off stage</span>
            <span className="boot"> â€¢ <a href="#">Boot</a></span>
          </div>
        </div>
      </div>;


    if (participant.inSession) {
      sessionParticipants.push(html);
    } else {
      leftParcipants.push(html);
    }
  });

  return (
    <section className='list'>
      {sessionParticipants}
      {leftParcipants}
    </section>
  );
}

export default List;
