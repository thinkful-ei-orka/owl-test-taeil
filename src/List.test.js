import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import List from './List'

describe(`List Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    expect(
      renderer.create(<List />).toJSON()
    ).toMatchSnapshot()
  })
})
