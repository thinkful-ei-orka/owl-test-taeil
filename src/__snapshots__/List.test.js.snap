// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders the UI as expected 1`] = `
<section
  className="list"
>
  <div
    className="participant session-true stage-true"
  >
    <img
      alt="Koren Templeton avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div>
      <h6>
        Koren Templeton
      </h6>
      <div>
        <span
          className="status"
        />
        <span
          className="on-stage"
        >
          on stage
        </span>
        <span
          className="off-stage"
        >
          off stage
        </span>
        <span
          className="boot"
        >
           • 
          <a
            href="#"
          >
            Boot
          </a>
        </span>
      </div>
    </div>
  </div>
  <div
    className="participant session-true stage-false"
  >
    <img
      alt="Caty Flucker avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div>
      <h6>
        Caty Flucker
      </h6>
      <div>
        <span
          className="status"
        />
        <span
          className="on-stage"
        >
          on stage
        </span>
        <span
          className="off-stage"
        >
          off stage
        </span>
        <span
          className="boot"
        >
           • 
          <a
            href="#"
          >
            Boot
          </a>
        </span>
      </div>
    </div>
  </div>
  <div
    className="participant session-true stage-false"
  >
    <img
      alt="Frank Runciman avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div>
      <h6>
        Frank Runciman
      </h6>
      <div>
        <span
          className="status"
        />
        <span
          className="on-stage"
        >
          on stage
        </span>
        <span
          className="off-stage"
        >
          off stage
        </span>
        <span
          className="boot"
        >
           • 
          <a
            href="#"
          >
            Boot
          </a>
        </span>
      </div>
    </div>
  </div>
  <div
    className="participant session-true stage-true"
  >
    <img
      alt="Ashla Attwool avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div>
      <h6>
        Ashla Attwool
      </h6>
      <div>
        <span
          className="status"
        />
        <span
          className="on-stage"
        >
          on stage
        </span>
        <span
          className="off-stage"
        >
          off stage
        </span>
        <span
          className="boot"
        >
           • 
          <a
            href="#"
          >
            Boot
          </a>
        </span>
      </div>
    </div>
  </div>
  <div
    className="participant session-false stage-false"
  >
    <img
      alt="Axe Kubicka avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div>
      <h6>
        Axe Kubicka
      </h6>
      <div>
        <span
          className="status"
        />
        <span
          className="on-stage"
        >
          on stage
        </span>
        <span
          className="off-stage"
        >
          off stage
        </span>
        <span
          className="boot"
        >
           • 
          <a
            href="#"
          >
            Boot
          </a>
        </span>
      </div>
    </div>
  </div>
</section>
`;
